docs/source/configuration/traffic-shaping.mdx

```diff
@@ -44,7 +33,135 @@ traffic_shaping:
         ttl: 10s # for each successful request, we register a token, that expires according to this option (default: 10s)
         retry_percent: 0.2 # defines the proportion of available retries to the current number of tokens
         retry_mutations: false # allows retries on mutations. This should only be enabled if mutations are idempotent
+```
+
+## Client side traffic shaping
+
+### Rate limiting
+
+The Apollo Router can apply rate limiting on client requests, as follows:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  router: # Rules applied to requests from clients to the router
+    global_rate_limit: # Accept a maximum of 10 requests per 5 secs. Excess requests must be rejected.
+      capacity: 10
+      interval: 5s # Must not be greater than 18_446_744_073_709_551_615 milliseconds and not less than 0 milliseconds
+```
+
+This rate limiting applies to all requests, there is no filtering per IP or other criterion.
+
+### Timeout
+
+The Apollo Router applies a default limit of 30 seconds to receive the entire client request. That limit is configurable:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  router: 
+    timeout: 50s # If a request to the router takes more than 50secs then cancel the request (30 sec by default)
+```
+
+### Compression
+
+Compression is automatically supported on the client side, depending on the `Accept-Encoding` header provided by the client.
+
+## Subgraph side traffic shaping
+
+The Apollo Router supports various options affecting traffic with subgraphs, that can either be defined for all subgraphs,
+or overriden per subgraph:
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    deduplicate_query: true # Enable query deduplication for all subgraphs.
+  subgraphs: # Rules applied to requests from the router to individual subgraphs
+    products:
+      deduplicate_query: false # Disable query deduplication for the products subgraph.
+```
+
+### Compression
+
+The Apollo Router can compress request bodies to subgraphs (along with response bodies to clients).
+It currently supports these algorithms: `gzip`, `br`, and `deflate`.
+
+```yaml title="router.yaml"
+traffic_shaping:
+  all:
+    compression: br # Enable brotli compression for all subgraphs.
```

Comment:
Is anything enabled by default here?

Reply:
it's not enabled by default, but now I see that decompressing the subgraph response will only work if we compress the subgraph request https://github.com/apollographql/router/blob/dev/apollo-router/src/plugins/traffic_shaping/mod.rs#L353-L357
I think we should always accept compressed subgraph responses,  right?

Reply:
opening https://github.com/apollographql/router/issues/2415 about this